sdk/storage/Azure.Storage.Blobs/src/PartitionedUploader.cs

```diff
@@ -227,7 +227,7 @@ internal class PartitionedUploader
                     accessTier,
                     cancellationToken);
             }
-            catch (Exception ex)
+            catch (Exception ex) when (BlobErrors.IsDiagnosticFailure(ex))
```

Comment:
Wonder if we can put this logic onto the ClientDiagnostics

Reply:
I think the logic is pretty Blobs specific.  If you want to have some other way of hooking it into `ClientDiagnostics` (maybe overriding a method in a derived `BlobsClientDiagnostics`), that could work too.